diff --git a/recipes/onnxruntime/all/conanfile.py b/recipes/onnxruntime/all/conanfile.py
index 0757ecdc0..bb8cca8bf 100644
--- a/recipes/onnxruntime/all/conanfile.py
+++ b/recipes/onnxruntime/all/conanfile.py
@@ -34,6 +34,11 @@ class OnnxRuntimeConan(ConanFile):
         "fPIC": True,
         "with_xnnpack": False,
         "with_cuda": False,
+        # Prevents libcurl dependency (fails to build due to system zlib)
+        # NOTE:
+        # 'use_system_tz_db' is specific to date/3.0.1 (version pinned by onnxruntime/1.18.1).
+        # Newer date recipes would need the tz_db=system option instead.
+        "date/*:use_system_tz_db": True,
     }
     short_paths = True
 
@@ -47,7 +52,7 @@ class OnnxRuntimeConan(ConanFile):
     def _compilers_minimum_version(self):
         return {
             "Visual Studio": "17",
-            "msvc": "193",
+            "msvc": "192",
             "gcc": "9",
             "clang": "5",
             "apple-clang": "10",
