--- a/cmake/adjust_global_compile_flags.cmake
+++ b/cmake/adjust_global_compile_flags.cmake
@@ -226,13 +226,17 @@ endmacro()
 
 #Set global compile flags for all the source code(including third_party code like protobuf)
 #This section must be before any add_subdirectory, otherwise build may fail because /MD,/MT mismatch
+message(STATUS "MSVC=${MSVC} | CMAKE_VS_PLATFORM_NAME=${CMAKE_VS_PLATFORM_NAME} | CMAKE_C_COMPILER_ARCHITECTURE_ID=${CMAKE_C_COMPILER_ARCHITECTURE_ID} | CMAKE_SYSTEM_PROCESSOR=${CMAKE_SYSTEM_PROCESSOR}")
 if (MSVC)
   if (CMAKE_VS_PLATFORM_NAME)
     # Multi-platform generator
     set(onnxruntime_target_platform ${CMAKE_VS_PLATFORM_NAME})
+  elseif (CMAKE_C_COMPILER_ARCHITECTURE_ID)
+    set(onnxruntime_target_platform ${CMAKE_C_COMPILER_ARCHITECTURE_ID})
   else()
     set(onnxruntime_target_platform ${CMAKE_SYSTEM_PROCESSOR})
   endif()
+  message(STATUS "Temporary onnxruntime_target_platform=${onnxruntime_target_platform}")
   if (onnxruntime_target_platform STREQUAL "ARM64")
     set(onnxruntime_target_platform "ARM64")
     enable_language(ASM_MARMASM)
@@ -241,18 +245,19 @@ if (MSVC)
   elseif (onnxruntime_target_platform STREQUAL "ARM" OR CMAKE_GENERATOR MATCHES "ARM")
     set(onnxruntime_target_platform "ARM")
     enable_language(ASM_MARMASM)
-  elseif (onnxruntime_target_platform STREQUAL "x64" OR onnxruntime_target_platform STREQUAL "x86_64" OR onnxruntime_target_platform STREQUAL "AMD64" OR CMAKE_GENERATOR MATCHES "Win64")
+  elseif (onnxruntime_target_platform MATCHES "[Xx]64" OR onnxruntime_target_platform MATCHES "[Xx]86_64" OR onnxruntime_target_platform STREQUAL "AMD64" OR CMAKE_GENERATOR MATCHES "Win64")
     set(onnxruntime_target_platform "x64")
     enable_language(ASM_MASM)
-  elseif (onnxruntime_target_platform STREQUAL "Win32" OR onnxruntime_target_platform STREQUAL "x86" OR onnxruntime_target_platform STREQUAL "i386" OR onnxruntime_target_platform STREQUAL "i686")
+  elseif (onnxruntime_target_platform STREQUAL "Win32" OR onnxruntime_target_platform MATCHES "[Xx]86" OR onnxruntime_target_platform STREQUAL "i386" OR onnxruntime_target_platform STREQUAL "i686")
     set(onnxruntime_target_platform "x86")
     enable_language(ASM_MASM)
     message("Enabling SAFESEH for x86 build")
     set(CMAKE_ASM_MASM_FLAGS "${CMAKE_ASM_MASM_FLAGS} /safeseh")
   else()
-    message(FATAL_ERROR "Unknown CMAKE_SYSTEM_PROCESSOR: ${CMAKE_SYSTEM_PROCESSOR}")
+    message(FATAL_ERROR "Unknown onnxruntime_target_platform: ${onnxruntime_target_platform}")
   endif()
 
+  message(STATUS "Final onnxruntime_target_platform=${onnxruntime_target_platform}")
 
   #Always enable exception handling, even for Windows ARM
   if (NOT onnxruntime_DISABLE_EXCEPTIONS)
